libs/designsystem/accordion/src/accordion.directive.ts

```diff
@@ -1,8 +1,44 @@
-import { Directive } from '@angular/core';
+import {
+  AfterContentInit,
+  ContentChildren,
+  Directive,
+  Input,
+  OnDestroy,
+  QueryList,
+} from '@angular/core';
+
+import { Subscription } from 'rxjs';
+
+import { AccordionItemComponent } from './accordion-item.component';
 
 @Directive({
   selector: '[kirbyAccordion], kirby-accordion',
   exportAs: 'kirby-accordion',
   standalone: false,
 })
-export class AccordionDirective {}
+export class AccordionDirective implements AfterContentInit, OnDestroy {
+  @Input() headingLevel: number;
+  @ContentChildren(AccordionItemComponent) accordionChildren: QueryList<AccordionItemComponent>;
+
+  private childrenChangesSub?: Subscription;
+
+  ngAfterContentInit(): void {
+    this.setHeadingLevels();
+
+    this.childrenChangesSub = this.accordionChildren.changes.subscribe(() => {
+      this.setHeadingLevels();
+    });
+  }
+
+  private setHeadingLevels(): void {
+    if (this.accordionChildren) {
```

Comment:
Consider only doing the work if headingLevel is set
```suggestion
    if (this.accordionChildren && this.headingLevel) {
```

Reply:
Additionally: Most likely, accordionChildren is always there, but might be length 0, so it should probably check for length > 0 instead of simply this.accordionChildren.